Add user to database

Sam09040 opened the pull request:
Using prisma client functions to add and find users

Sam09040 committed:
Mutation working

add:
- schema.graphql with the schema
- user interfaces
- resolver with mutation

change:
- prisma schema to match User schema
- '@apollo/server' for 'apollo-server' for  easier management of project

Sam09040 committed:
Working user add to bd

## add:
- password hash
- prisma client functions
- many users and user by id queries

Sam09040 committed:
Working user add to bd

## add:
- password hash
- prisma client functions
- many users and user by id queries

Sam09040 committed:
Add password validation

## add:
- password validation function and add it to the resolver

Sam09040 committed:
Add password validation

## add:
- password validation function and add it to the resolver

Sam09040 committed:
Remove password field from schema

Sam09040 committed:
Merge pull request #11 from Sam09040/feature/setup-user-mutation

Setup Mutation

sophia-bergamo commented on src/index.ts:
```diff
@@ -1,34 +1,87 @@
+
+            const { name, email, password, birthDate } = data;
+
+            const existingEmail = await prisma.user.findFirst({
```
como o e-mail é único no banco, é mais eficiente usar a função que procura por valores únicos ao invés de procurar o primeiro

Sam09040 replied:
resolvido!

Sam09040 committed:
Small changes

Sam09040 committed:
Small changes

Sam09040 committed:
File organization

## add:
- graphql folder under src
- schema.graphql under graphql
- resolvers.ts under graphql

## update:
- small change on index.ts

Sam09040 committed:
File organization

## add:
- graphql folder under src
- schema.graphql under graphql
- resolvers.ts under graphql

## update:
- small change on index.ts

Sam09040 committed:
Merge branch 'feature/db-user-add' of https://github.com/Sam09040/botActivities into feature/db-user-add
